UnityProject/Assets/Scripts/Objects/Climbable.cs

```diff
@@ -0,0 +1,72 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Climbable : MonoBehaviour, ICheckedInteractable<MouseDrop>
+{
+
+	public float ClimbTime = 1.0f;
+
+
+	public bool WillInteract(MouseDrop interaction, NetworkSide side)
+	{
+		if (!DefaultWillInteract.Default(interaction, side)) return false;
+
+		PlayerSync playerSync;
+		CustomNetTransform netTransform;
+		if(interaction.UsedObject.TryGetComponent(out playerSync))
+		{
+			if (playerSync.IsMoving || playerSync.playerMove.IsBuckled)
+			{
+				return false;
+			}
+
+			// Do a sanity check to make sure someone isn't dropping the shadow from like 9000 tiles away.
+			var mag = (interaction.TargetObject.transform.position - playerSync.ServerPosition).magnitude;
+			if (mag > PlayerScript.interactionDistance)
```

Comment:
move the magnitud check to a private method so you can reuse it.

Reply:
Pretty sure `if (!DefaultWillInteract.Default(interaction, side)) return false;` will check the distance, though double check

Reply:
yeah, that makes sense. Maybe Max found that check in our current implementation of climbing tables? I know we had some issues with distance before.

Reply:
No it doesn't.